ethereum/gnosis_protocol_v2/view_solvers.sql

```diff
@@ -1,64 +1,77 @@
+                 (decode('97dd6a023b06ba4722aF8Af775ec3C2361e66684', 'hex'), 'barn', '0x'),
+                 (decode('6372bcbf66656e91b9213b61d861b5e815296207', 'hex'), 'barn', 'ParaSwap'),
+                 (decode('158261d17d2983b2cbaecc1ae71a34617ae4cb16', 'hex'), 'barn', 'MIP'),
+                 (decode('8c9d33828dace1eb9fc533ffde88c4a9db115061', 'hex'), 'barn', '1inch'),
+                 (decode('bfaf2b5e351586551d8bf461ba5b2b5455b173da', 'hex'), 'barn', 'Baseline'),
+                 (decode('b8650702412d0aa7f01f6bee70335a18d6a78e77', 'hex'), 'barn', 'Naive'),
+                 (decode('583cD88b9D7926357FE6bddF0E8950557fcDA0Ca', 'hex'), 'barn', 'DexCowAgg'),
+                 (decode('6c2999b6b1fad608ecea71b926d68ee6c62beef8', 'hex'), 'barn', 'Legacy'),
+                 (decode('ED94b86275447e28ddbDd17BBEB1f62D607b5119', 'hex'), 'barn', 'Legacy'),
+                 (decode('8ccc61dba297833dbe5d95fd6360f106b9a7576e', 'hex'), 'barn', 'Naive'),
+                 (decode('0d2584da2f637805071f184bcfa1268ebae8a24a', 'hex'), 'barn', 'Baseline'),
+                 (decode('a0044c620da7f2876da7004719b8370eb7be5e50', 'hex'), 'barn', 'MIP'),
+                 (decode('da324c2f06d3544e7965767ce9ca536dcb67a660', 'hex'), 'barn', 'Quasimodo'),
+                 (decode('e33062a24149f7801a48b2675ed5111d3278f0f5', 'hex'), 'barn', '1Inch'),
+                 (decode('080a8b1e2f3695e179453c5e617b72a381be44b9', 'hex'), 'barn', 'ParaSwap'),
+                 (decode('de786877a10dbb7eba25a4da65aecf47654f08ab', 'hex'), 'barn', '0x'),
+                 (decode('dae69affe582d36f330ee1145995a53fab670962', 'hex'), 'barn', 'CowDexAg'),
+                 (decode('0b78e29ee55aa73b366730cf512d65c514eeb196', 'hex'), 'barn', 'BalancerSOR'),
+                 (decode('22dee0935c77d32c7241362b14e76fc2d5ef657d', 'hex'), 'barn', 'BalancerSOR'),
+                 (decode('0798540ee03a8c2e68cef19c56d1faa86271d5cf', 'hex'), 'service', 'Withdraw'),
+                 (decode('df30c9502eafea21ecc8410108dda338dd5047c5', 'hex'), 'service', 'Withdraw'),
+                 (decode('256bb5ad3dbdf61ae08d7cbc0b9223ccb1c60aae', 'hex'), 'service', 'Withdraw'),
+                 (decode('84e5c8518c248de590d5302fd7c32d2ae6b0123c', 'hex'), 'service', 'Withdraw'),
+                 (decode('2caef7f0ee82fb0abf1ab0dcd3a093803002e705', 'hex'), 'test', 'Test Solver 1'),
+                 (decode('56d4ed5e49539ebb1366c7d6b8f2530f1e4fe753', 'hex'), 'test', 'Test Solver 2')
          ) as _
-),
-
-     unknown_solvers as (
-         select distinct(solver), 'new', True, 'Uncatalogued'
-         from gnosis_protocol_v2."GPv2Settlement_evt_Settlement"
-         where solver not in (select address from known_solvers)
-     )
-
-SELECT *
-FROM (
-         select *
-         from known_solvers
-         union
-         select *
-         from unknown_solvers
-     ) as _;
+)
+-- Combining the metadata with activation status for final table
```

Comment:
A little bit above, you are introducing  block numbers, when the solvers get's activated and deactived.
```
    select solver, evt_block_number, evt_index, True as activated
    from gnosis_protocol_v2."GPv2AllowListAuthentication_evt_SolverAdded"
    union
    select solver, evt_block_number, evt_index, False as activated
    from gnosis_protocol_v2."GPv2AllowListAuthentication_evt_SolverRemoved"
    ```
 But here you just store the final end result. Hence, a deactivated solver will have two rows: activated and deactivated, without the block number info. Is this useful in some sense? Or should the block number also be in this table.

Reply:
I think you might be missing the next sub query which selects the most recent event related to a distinct solver where we choose the relevant boolean value to determine if the solver is active or not active.

Reply:
```suggestion
-- Combining the metadata with current activation status for final table
```